### ISSUE TITLE
Question about causal attention mask in SASRec backbone

### ISSUE THREAD

**Issue by tomatoboy-hub** [liuqidong07/LLM4CDSR-pytorch#2]
Hi @liuqidong07 
I have a question regarding the SASRec backbone here:
https://github.com/liuqidong07/LLM4CDSR-pytorch/blob/859ab23009074732138f1fc1eae75f504bae7c7f/models/SASRec.py#L51

In the self-attention layer, it seems that attn_mask (causal mask) is not explicitly passed. I was wondering if there is a specific reason for this design.
I would appreciate it if you could clarify this. Thanks!
